common/util.c

```diff
@@ -86,8 +84,17 @@ bool read_ed25519_key(uint8_t *in, size_t in_len, uint8_t *out,
   }
 
   int ret;
-  BIO *b64 = BIO_new(BIO_f_base64());
-  BIO *bio = BIO_new(BIO_s_mem());
+  BIO *b64 = NULL;
+  BIO *bio = NULL;
+
+  b64 = BIO_new(BIO_f_base64());
+  if (b64 == NULL) {
+    return false;
+  }
+  bio = BIO_new(BIO_s_mem());
+  if (bio == NULL) {
```

Comment:
this construct still leaks if the first operation succeeds and the second fails.

Reply:
It should be sufficient to call Bio_free(b64) if the second operation fails, right?

Reply:
yes, I think so

Reply:
Fixed